src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -252,10 +256,12 @@ public static BoundConversion SynthesizedNonUserDefined(SyntaxNode syntax, Bound
                 isBaseConversion: false,
                 @checked: @checked,
                 explicitCastInCode: explicitCastInCode,
+                isExplicitlyNullable: isExplicitlyNullable,
                 constantValueOpt: constantValueOpt,
                 type: type,
                 hasErrors: hasErrors || !conversion.IsValid)
         {
+            Debug.Assert(explicitCastInCode || !isExplicitlyNullable);
```

Comment:
>Debug.Assert(explicitCastInCode || !isExplicitlyNullable); [](start = 12, length = 58)

This probably belongs in a Validate method that is called from the root constructor. Generator can declare and inject that call for VB, it might handle C# as well. See ``` HasValidate="true"``` on BoundNodes.xml

Reply:
Thanks. However, it looks like `BoundNodeClassWriter` only uses `HasValidate` in VB currently.